gsa/src/web/graphql/wizards.js

```diff
@@ -66,8 +68,108 @@ export const useRunQuickFirstScan = () => {
         });
       });
     },
-    [createTarget, createTask, startTask, sessionTimeout, userTimezone],
+    [createTarget, createTask, startTask],
   );
 
   return [runQuickFirstScan];
 };
+
+export const useRunModifyTask = () => {
+  const [createAlert] = useCreateAlert();
+  const [createSchedule] = useCreateSchedule();
+  const [modifyTask] = useModifyTask();
+
+  const runModifyTask = useCallback(
+    ({
+      alert_email,
+      start_date,
+      start_minute,
+      start_hour,
+      start_timezone,
+      reschedule,
+      tasks,
+    }) => {
+      start_date.hours(start_hour);
+      start_date.minutes(start_minute);
+
+      const [task] = tasks;
+      const {name, id: taskId, alerts} = task;
+
+      const rightNow = new Date().toISOString();
+
+      const event = Event.fromData(
+        {startDate: start_date},
+        start_timezone,
+        false, // if true, then the event will be scheduled at the given time, but at the user timezone instead. If I create an event at 22:00 Shanghai time, it will always be scheduled at 22:00 CEST (User timezone)
```

Comment:
As mentioned already I am really sure this is wrong. the passed startDate needs to be already in the correct timezone.